### PR TITLE
add fields to subclasses Bulldozer, Excavator, Truck;

### PR DESCRIPTION
**Pull request by aksoli666**
create BulldozerProducer, ExcavatorProducer, TruckProducer; configure parameter MachineProducer;
change signature methods and realized them:
getAll(), fill(), startWorking();

### CODE REVIEW

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 1]
```diff
@@ -2,37 +2,10 @@
 
 import java.util.List;
 
-/**
- * Parameterize this service and add its implementation in a separate class.
- */
-public interface MachineService {
+public interface MachineService<T> {
+    List<core.mate.academy.model.Machine> getAll(Class<? extends T> type);
```
Comment (kerrrusha) [inline src/main/java/core/mate/academy/service/MachineService.java:6]:
```suggestion
    List<T> getAll(Class<? extends T> type);
```

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 2]
```diff
@@ -2,37 +2,10 @@
...
-     * Return the list of machines.
-     * In the implementation of this method please use your MachineProducer implementations
-     * See that 'Class type' is not parametrized.
-     * Consider to parametrize this Class< PARAMETRIZE_ME > with specific wildcard bounding
-     *
-     * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
-     * @return the list of machines
-     */
-    List<Object> getAll(Class type);
+    void fill(List<? super T> machines, Object value);
```
Comment (kerrrusha) [inline src/main/java/core/mate/academy/service/MachineService.java:8]:
```suggestion
    void fill(List<? super T> machines, T value);
```

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 3]
```diff
@@ -1,7 +1,46 @@
...
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            BulldozerProduce bulldozerProduce = new BulldozerProduce();
```
Comment (kerrrusha) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:14]: make it `private final` field

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 4]
```diff
@@ -1,7 +1,46 @@
...
+            List<? extends Machine> bulldozers = bulldozerProduce.get();
+            return new ArrayList<>(bulldozers);
+        }
+
+        if (type == Excavator.class) {
+            ExcavatorProduce excavatorProduce = new ExcavatorProduce();
```
Comment (kerrrusha) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:20]: make it `private final` field

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 5]
```diff
@@ -1,7 +1,46 @@
...
+            List<Excavator> excavators = excavatorProduce.get();
+            return new ArrayList<>(excavators);
+        }
+
+        if (type == Truck.class) {
+            TruckProduce truckProduce = new TruckProduce();
```
Comment (kerrrusha) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:26]: make it `private final` field

File: src/main/java/core/mate/academy/service/BulldozerProduce.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulldozerProduce implements MachineProducer<Bulldozer> {
```
Comment (kerrrusha) [inline src/main/java/core/mate/academy/service/BulldozerProduce.java:7]:
```suggestion
public class BulldozerProducer implements MachineProducer<Bulldozer> {
```

File: src/main/java/core/mate/academy/service/ExcavatorProduce.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProduce implements MachineProducer<Excavator> {
```
Comment (kerrrusha) [inline src/main/java/core/mate/academy/service/ExcavatorProduce.java:7]:
```suggestion
public class ExcavatorProducer implements MachineProducer<Excavator> {
```

File: src/main/java/core/mate/academy/service/TruckProduce.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProduce implements MachineProducer<Truck> {
```
Comment (kerrrusha) [inline src/main/java/core/mate/academy/service/TruckProduce.java:7]:
```suggestion
public class TruckProducer implements MachineProducer<Truck> {
```

**Review by koval-kn — CHANGES_REQUESTED** [review]

**Review by Rommelua — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

**Review by koval-kn — APPROVED** [review]

**Pull request closed without merging**